feature_flag.go

```diff
@@ -262,8 +265,38 @@ func retrieveFlagsAndUpdateCache(config Config, cache cache.Manager, retrieverMa
 			newFlags[flagName] = value
 		}
 	}
+	return newFlags, nil
+}
+
+// retrieveFlagsAndInitializeCache is called when the feature flag is initialized
+func retrieveFlagsAndInitializeCache(config Config, cache cache.Manager, retrieverManager *retriever.Manager) error {
```

Comment:
Rather having almost the same function for `retrieveFlagsAndInitializeCache` and `retrieveFlagsAndUpdateCache` I would have prefer to add a parameter to the previous function `retrieveFlagsAndUpdateCache` and use the same code on both side.

Something like 
```go
func retrieveFlagsAndUpdateCache(config Config, cache cache.Manager, retrieverManager *retriever.Manager, isInitialization bool) error {

// ...

    if isInitialization && config.DisableNotificationOnInit {
		err = cache.UpdateCache(newFlags, config.internalLogger)
	} else {
		err = cache.UpdateCacheAndNotify(newFlags, config.internalLogger)
	}

```

Reply:
Same deal as here: https://github.com/thomaspoignant/go-feature-flag/pull/2534#discussion_r1811545406

I think I just tend to prefer having explicit function/method names just to make it a bit more readable on the caller code and avoid introducing too many parameters in a single method, but I think either works. To make it consistent with the code on the above comment, let me change this one as well 👍